src/foam/dao/FileJournal.js

```diff
@@ -283,11 +286,18 @@ foam.CLASS({
         int successReading = 0;
         JSONParser parser = getParser();
 
-        try ( BufferedReader reader = getReader() ) {
+        try {
+          BufferedReader reader = new BufferedReader(new FileReader(getFile()));
+
           for ( String line ; ( line = reader.readLine() ) != null ; ) {
             if ( SafetyUtil.isEmpty(line)        ) continue;
             if ( COMMENT.matcher(line).matches() ) continue;
 
+            if ( ! SafetyUtil.isEmpty(getPrefix()) ) {
+              if ( ! line.startsWith(getPrefix()) ) continue;
```

Comment:
So this would mean that we re-reading the entire journal file for every DAO? O(nm) : n is the number for lines in the journal and m is the total number of DAOs?